storage/innobase/include/lock0lock.h

```diff
@@ -1020,6 +1020,9 @@ lock_rtr_move_rec_list(
 						moved */
 	ulint			num_move);	/*!< in: num of rec to move */
 
+/** Cancel a waiting lock request and release possibly waiting transactions */
+void lock_cancel_waiting_and_release(lock_t *lock);
```

Comment:
This should be inside `#ifdef WITH_WSREP`.

Maybe even better, an `#ifdef WITH_WSREP` member function could be added to `lock_sys_t`, or `lock_wait_wsrep_kill()` could be moved to `lock0lock.cc` and `find_thread_by_id()` be made available to that compilation unit somehow.